From: FUJITA Tomonori <fujita.tomonori@lab.ntt.co.jp>
Date: Sat, 3 Jul 2010 17:45:40 +0900
Subject: [PATCH] block: remove q->prepare_flush_fn completely
Git-commit: 00fff26539bfe3fad21c164fc4002d9ede056fb0
References: FATE#311692
Patch-Mainline: 2.6.36

This removes q->prepare_flush_fn completely (changes the
blk_queue_ordered API).

Signed-off-by: FUJITA Tomonori <fujita.tomonori@lab.ntt.co.jp>
Reviewed-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Jens Axboe <jaxboe@fusionio.com>
Signed-off-by: Hannes Reinecke <hare@suse.de>

Automatically created from "patches.suse/block-remove-q-prepare_flush_fn-completely.patch" by xen-port-patches.py

--- sle11sp2-2011-06-14.orig/drivers/xen/blkfront/vbd.c	2011-05-31 11:19:19.000000000 +0200
+++ sle11sp2-2011-06-14/drivers/xen/blkfront/vbd.c	2011-06-14 16:41:53.000000000 +0200
@@ -476,7 +476,7 @@ xlvbd_barrier(struct blkfront_info *info
 	int err;
 
 	err = blk_queue_ordered(info->rq,
-		info->feature_barrier ? QUEUE_ORDERED_DRAIN : QUEUE_ORDERED_NONE, NULL);
+		info->feature_barrier ? QUEUE_ORDERED_DRAIN : QUEUE_ORDERED_NONE);
 	if (err)
 		return err;
 	pr_info("blkfront: %s: barriers %s\n",
--- sle11sp2-2011-06-14.orig/drivers/xen/blktap2-new/device.c	2011-02-24 16:23:08.000000000 +0100
+++ sle11sp2-2011-06-14/drivers/xen/blktap2-new/device.c	2011-06-14 16:41:49.000000000 +0200
@@ -303,7 +303,7 @@ blktap_device_configure(struct blktap *t
 	blk_queue_dma_alignment(rq, 511);
 
 	/* We are reordering, but cacheless. */
-	blk_queue_ordered(rq, QUEUE_ORDERED_DRAIN, NULL);
+	blk_queue_ordered(rq, QUEUE_ORDERED_DRAIN);
 
 	spin_unlock_irq(&dev->lock);
 }

